code/Helper/Data.php

```diff
@@ -511,6 +512,65 @@ public function rebuildStoreProductIndexPage($storeId, $collectionDefault, $page
         $this->logger->stop('rebuildStoreProductIndexPage '.$this->logger->getStoreName($storeId).' page ' . $page . ' pageSize '. $pageSize);
     }
 
+    /**
+     * Removes products that are no longer indexable, i.e. they are disabled,
+     * from the specified Algolia index. You can provide an array from product
+     * IDs that you want to query. Otherwise, the entire product collection will
+     * be queried.
+     *
+     * @param string $storeId
+     * @param array $productIds
+     */
+    public function removeNonIndexableProducts($storeId, array $productIds = array())
+    {
+        $this->logger->start('REMOVE FROM ALGOLIA');
+
+        if (!$this->config->getShowOutOfStock()) {
+            $nonIndexableProductIds = $this->getNonIndexableProductIds($storeId, $productIds);
+            if (count($nonIndexableProductIds) > 0) {
+                $this->removeProducts($nonIndexableProductIds, $storeId);
+            }
+        }
+
+        $this->logger->stop('REMOVE FROM ALGOLIA');
+    }
+
+    /**
+     * Returns an array of the IDs of products that are no longer indexable,
+     * i.e. they are disabled. You can provide an array from product IDs that
+     * you want to query. Otherwise, the entire product collection will be
+     * queried.
+     *
+     * @param string $storeId
+     * @param array $productIds
+     * @return array
+     */
+    public function getNonIndexableProductIds($storeId, array $productIds = array())
+    {
+        $nonIndexableProductIds = array();
+
+        /** @var Mage_Catalog_Model_Resource_Product_Collection $products */
+        $products = Mage::getModel('catalog/product')->getCollection();
+        $products
+            ->addStoreFilter($storeId)
+            ->addAttributeToFilter('type_id', 'configurable')
```

Comment:
Uh yeah, fair point. Algolia is pretty expensive so I didn't want to process all of the simples since that's like 30,000 requests for us. I guess I can just disable products that are not visible online!